geneLevel.R

```diff
@@ -172,28 +211,150 @@ get_gene_level_calls <- function(cncf_files,
+    exome_bed <- suppressWarnings(fread(paste0('grep -v "^@" ',getSDIR(),"/Homo_sapiens.GRCh37.75.canonical_exons.bed")))
+    setnames(exome_bed, c("chr", "start", "end", "name","blank","strand"))
+    setkey(exome_bed, chr, start, end)
+    cross <- foverlaps(exome_bed, concat_cncf_txt, nomatch=NA)
+    cross <- cross[!is.na(ID)]
+    cross[,Hugo_Symbol:=gsub(":.*$", "", name)]
+    genecount <- cross[!Hugo_Symbol %in% c("Tiling", "FP", "intron"),
+                            list(chr = unique(chr),
+                                 seg.start=unique(loc.start),
+                                 seg.end=unique(loc.end),
+                                 frac_elev_major_cn=unique(frac_elev_major_cn),
+                                 Nprobes = .N),
+                            keyby=list(Tumor_Sample_Barcode, Hugo_Symbol, tcn=tcn, lcn=lcn, cf=cf,
+                                       tcn.em = tcn.em, lcn.em = lcn.em, cf.em = cf.em)]
+    ### fix bug where lcn == NA even when tcn is 1
+    genecount[tcn == 1 & is.na(lcn), lcn := 0]
+    ### apply WGD threshold
+    genecount[, WGD := factor(ifelse(frac_elev_major_cn > WGD_threshold, "WGD", "no WGD"))]
+    setkey(genecount, Tumor_Sample_Barcode, Hugo_Symbol)
+    ### annotate integer copy number
+    genecount <- annotate_integer_copy_number(genecount, amp_threshold)
+    ### remove duplicate entries for partial deletions
+    ### the lower value is chosen on the basis that a
+    ### partial deletion is a deletion but a
+    ### partial amplification is not an amplification
+    genecount <- genecount[order(FACETS_CNA, -Nprobes)]
+    genecount <- genecount[!duplicated(genecount, by=c("Tumor_Sample_Barcode", "Hugo_Symbol"))]
+    genecount = genecount  %>% 
+      mutate(segid = str_c(Tumor_Sample_Barcode,';',chr,';',seg.start,';',seg.end))
+    seg.count = plyr::count(genecount, vars="segid") %>%
+                            mutate(count = freq) %>% select(-c(freq))
+    
+    genelevelcalls0 = dplyr::inner_join(genelevelcalls0,seg.count,by='segid')
+  
+    ### Step 3 SET columns needed for filters & APPLY filters
+    #genelevelcalls0= genelevelcalls0 %>% mutate(CFcut = plyr::mapvalues(Tumor_Sample_Barcode, gene_level$T, gene_level$CFcut))
+  
+    genelevelcalls0 = genelevelcalls0 %>% mutate(FACETS_CALL.ori = FACETS_CALL.em, 
+                           FACETS_CALL.em = ifelse( FACETS_CALL.em %in% c("AMP","AMP (LOH)","AMP (BALANCED)","HOMDEL"), 
+                                              ifelse( (FACETS_CALL.em %in% c("AMP","AMP (LOH)","AMP (BALANCED)") & seg.len < 10000000 & (tcn.em > 8 | count <=10 | cf.em > CFcut )), FACETS_CALL.em, 
+                                                ifelse( (FACETS_CALL.em == "HOMDEL" & seg.len < 10000000 & count <= 10), FACETS_CALL.em, "ccs_filter")), FACETS_CALL.em )) 
```

Comment:
Use variable `max_seg_length` here instead of hardcoding. Change its value above from `25000000` to `10000000`.

Reply:
we could use max_seg_length instead of hardcoding, we'll do that in the next version.

Reply:
I'll add an argument